src/test/java/com/shopify/stepDefinitions/AddToCartSteps.java

```diff
@@ -0,0 +1,93 @@
+package com.shopify.stepDefinitions;
+
+import com.shopify.pageObjects.CartPage;
+import com.shopify.pageObjects.CheckOutPage;
+import com.shopify.pageObjects.HomePage;
+import com.shopify.pageObjects.LogInPage;
+import com.shopify.utilityClasses.BrowserUtil;
+import com.shopify.utilityClasses.ConfigReader;
+import io.cucumber.java.en.*;
+import org.openqa.selenium.WebDriver;
+import org.testng.Assert;
+
+public class AddToCartSteps {
+
+    WebDriver driver;
+    LogInPage lp;
+    HomePage hp;
+    CartPage cp;
+    CheckOutPage chkpg;
+    String expectedAmount = "$29.99";
+    String orderConfMsg ="Your order has been dispatched, and will arrive just as fast as the pony can get there!";
+
+    @Given("User launches the browser")
+    public void user_launches_the_browser() {
+        driver = BrowserUtil.getDriver(ConfigReader.get("BROWSER"));
+        System.out.println(ConfigReader.get("BROWSER") + " browser launched!");
+        driver.manage().window().maximize();
+    }
+
+    @When("User opens URL")
+    public void user_opens_URL() {
+        driver.get(ConfigReader.get("BASE_URL"));
+    }
+
+    @And("User enters email and password")
+    public void user_enters_email_and_password() {
+        lp= new LogInPage(driver);
+        lp.setUserName(ConfigReader.get("USERNAME"));
+        lp.setPassword(ConfigReader.get("PASSWORD"));
+    }
+
+    @And("Clicks on Login")
+    public void clicks_on_login(){
+        lp.clickLoginBtn();
```

Comment:
This will throw null pointer exception if used before user_enters_email_and_password page should reassigned object value on each page.

Reply:
Made declaration of drivers privatised